CONTRIBUTING.md

```diff
@@ -1,3 +1,12 @@
+# Functional contribution
+
+We are welcome for any contribution. But every new feature implemented in this plugin should:
+ 
+- Be useful enough for lot of people (should not cover only your professional case)
+- Should not break existing use cases
+- Should be easily maintained (so maintainers need some time to think about architecture of implementation)
+- Have at least one test for positive use case
+
```

Comment:
Extra proposals:
- Should be compatible at least with the last LTS line
- Should pass the PR builder check (including unit tests, Findbugs and checkstyle)
- Should avoid breaking the backward compatibility in existing APIs. If the compatibility break is required, the new feature should be well justified.
   - See [this guide](https://wiki.eclipse.org/Evolving_Java-based_APIs_2) for particular examples
   - Jenkins provides specific solutions for [retaining the backward compatibility](https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility)